PhysicsTools/PatUtils/plugins/L1PrefiringWeightProducer.cc

```diff
@@ -0,0 +1,466 @@
+  iEvent.put(std::move(nonPrefiringProbJet), "nonPrefiringProbJet");
+  iEvent.put(std::move(nonPrefiringProbJetUp), "nonPrefiringProbJetUp");
+  iEvent.put(std::move(nonPrefiringProbJetDown), "nonPrefiringProbJetDown");
+
+  auto nonPrefiringProbPhoton = std::make_unique<double>(nonPrefiringProbaPhoton[0]);
+  auto nonPrefiringProbPhotonUp = std::make_unique<double>(nonPrefiringProbaPhoton[1]);
+  auto nonPrefiringProbPhotonDown = std::make_unique<double>(nonPrefiringProbaPhoton[2]);
+  iEvent.put(std::move(nonPrefiringProbPhoton), "nonPrefiringProbPhoton");
+  iEvent.put(std::move(nonPrefiringProbPhotonUp), "nonPrefiringProbPhotonUp");
+  iEvent.put(std::move(nonPrefiringProbPhotonDown), "nonPrefiringProbPhotonDown");
+
+  auto nonPrefiringProbMuon = std::make_unique<double>(nonPrefiringProbaMuon[0]);
+  auto nonPrefiringProbMuonUp = std::make_unique<double>(nonPrefiringProbaMuon[1]);
+  auto nonPrefiringProbMuonDown = std::make_unique<double>(nonPrefiringProbaMuon[2]);
+  iEvent.put(std::move(nonPrefiringProbMuon), "nonPrefiringProbMuon");
+  iEvent.put(std::move(nonPrefiringProbMuonUp), "nonPrefiringProbMuonUp");
+  iEvent.put(std::move(nonPrefiringProbMuonDown), "nonPrefiringProbMuonDown");
+}
+
+double L1PrefiringWeightProducer::getPrefiringRate(double eta,
+                                                   double pt,
+                                                   TH2F* h_prefmap,
+                                                   fluctuations fluctuation) const {
+  if (h_prefmap == nullptr && !skipwarnings_)
+    std::cout << "Prefiring map not found, setting prefiring rate to 0 " << std::endl;
+  if (h_prefmap == nullptr)
+    return 0.;
+  //Check pt is not above map overflow
+  int nbinsy = h_prefmap->GetNbinsY();
+  double maxy = h_prefmap->GetYaxis()->GetBinLowEdge(nbinsy + 1);
+  if (pt >= maxy)
+    pt = maxy - 0.01;
+  int thebin = h_prefmap->FindBin(eta, pt);
+
+  double prefrate = h_prefmap->GetBinContent(thebin);
+
+  double statuncty = h_prefmap->GetBinError(thebin);
+  double systuncty = prefiringRateSystUnc_ * prefrate;
+
+  if (fluctuation == up)
+    prefrate = std::min(1., prefrate + sqrt(pow(statuncty, 2) + pow(systuncty, 2)));
+  if (fluctuation == down)
+    prefrate = std::max(0., prefrate - sqrt(pow(statuncty, 2) + pow(systuncty, 2)));
+  return prefrate;
+}
+double L1PrefiringWeightProducer::getPrefiringRateMuon(double eta,
+                                                       double phi,
+                                                       double pt,
+                                                       fluctuations fluctuation) const {
+  double prefrate;
+  double statuncty;
+  if ((dataeraMuon_.find("2016") != std::string::npos) && (eta > 1.24 && eta < 1.6) &&
+      (phi > 2.44346 && phi < 2.79253)) {
+    if (parametrizationHotSpot_ == nullptr && !skipwarnings_)
+      std::cout << "Prefiring parametrization not found, setting prefiring rate to 0 " << eta << " " << phi
```

Comment:
Can you change the cout to LogError/Warning?

Reply:
Done.